zkevm-circuits/src/rlp_decoder_tables.rs

```diff
@@ -0,0 +1,764 @@
+                    for byte_val in 0..=255u8 {
+                        let (rlp_type, decodable) = decode_rule.rule_check(byte_val);
+                        let rule_table_row = [
+                            table_tag,
+                            *tx_type as u64,
+                            *tx_field_tag as u64,
+                            rlp_type as u64,
+                            byte_val as u64,
+                            decodable as u64,
+                        ];
+
+                        log::trace!("rule_table_row: {:?} @ offset {}.", &rule_table_row, offset);
+                        rule_table_row
+                            .iter()
+                            .zip([
+                                self.table_tag,
+                                self.tx_type,
+                                self.tx_field_tag,
+                                self.rlp_type,
+                                self.byte_0,
+                                self.decodable,
+                            ])
+                            .try_for_each(|(value, col)| {
+                                region
+                                    .assign_fixed(
+                                        || "load rlp rule decoder table",
+                                        col,
+                                        offset,
+                                        || Value::known(F::from(*value)),
+                                    )
+                                    .map(|_| ())
+                            })?;
+                        offset += 1;
+                    }
+                }
+                Ok(())
+            },
+        )
+    }
+}
+
+/// Table that contains the fields of possible state transitions
+#[derive(Clone, Debug)]
+pub struct TxFieldSwitchTable {
+    /// table tag of the table
+    pub table_tag: Column<Fixed>,
+    /// The current tx field
+    pub current_tx_field: Column<Fixed>,
+    /// The next tx field
+    pub next_tx_field: Column<Fixed>,
+}
+
+static TX_FIELD_TRANSITION_TABLE: [(RlpTxFieldTag, &[RlpTxFieldTag]); 15] = [
+    (
+        RlpTxFieldTag::TxListRlpHeader,
+        &[RlpTxFieldTag::TxRlpHeader, RlpTxFieldTag::DecodeError],
+    ),
+    (
+        RlpTxFieldTag::TxRlpHeader,
+        &[RlpTxFieldTag::Nonce, RlpTxFieldTag::DecodeError],
+    ),
+    (
+        RlpTxFieldTag::Nonce,
+        &[RlpTxFieldTag::GasPrice, RlpTxFieldTag::DecodeError],
+    ),
+    (
+        RlpTxFieldTag::GasPrice,
+        &[RlpTxFieldTag::Gas, RlpTxFieldTag::DecodeError],
+    ),
+    (
+        RlpTxFieldTag::Gas,
+        &[RlpTxFieldTag::To, RlpTxFieldTag::DecodeError],
+    ),
+    (
+        RlpTxFieldTag::To,
+        &[RlpTxFieldTag::Value, RlpTxFieldTag::DecodeError],
+    ),
+    (
+        RlpTxFieldTag::Value,
+        &[RlpTxFieldTag::Data, RlpTxFieldTag::DecodeError],
+    ),
+    (
+        RlpTxFieldTag::Data,
+        &[RlpTxFieldTag::Data, RlpTxFieldTag::DecodeError],
```

Comment:
just double-checking. This state transition is not in the documentation. It is actually missing from the docs or is this code invalid?

Reply:
Oh, you right! it's invalid transition, will remove.